### PR TITLE
Add dolt_workspace_* system tables

### PR DESCRIPTION
**Pull request by macneale4**
This is the read only addition of the `dolt_workspace_{table}` s. These dynamically generated tables are always relative to HEAD for the given session of the caller. There are no commits as a result, and the schema of the output looks like:

| ID (int) | STAGED (bool) | DIFF_TYPE (string) | to_A | to_B | ... | from_A | from_B | ... |

Currently there is no mechanism to update this table directly, but in the future that will make it possible to have fine grain modification of your workspace, similar to `git add --patch`

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>88b8ab1</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>88b8ab1</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>dc1c9bc</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>dc1c9bc</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>e81895f</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>e81895f</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>a598725</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>a598725</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+	{
+		Name: "dolt_workspace_* multiple edits of a single row",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"call dolt_commit('-Am', 'creating table t');",
+
+			"insert into tbl values (42,42);",
+			"insert into tbl values (43,43);",
+			"call dolt_commit('-am', 'inserting rows 3 rows at HEAD');",
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go:32]:
```suggestion
			"call dolt_commit('-am', 'inserting 2 rows at HEAD');",
```

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	},
+	{
+		Name: "dolt_workspace_* single unstaged row",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"call dolt_commit('-Am', 'creating table t');",
+
+			"insert into tbl values (42,42);",
+			"insert into tbl values (43,43);",
+			"call dolt_commit('-am', 'inserting rows 3 rows at HEAD');",
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go:73]:
```suggestion
			"call dolt_commit('-am', 'inserting 2 rows at HEAD');",
```

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	},
+	{
+		Name: "dolt_workspace_* inserted row",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"call dolt_commit('-Am', 'creating table t');",
+
+			"insert into tbl values (42,42);",
+			"insert into tbl values (43,43);",
+			"call dolt_commit('-am', 'inserting rows 3 rows at HEAD');",
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go:94]:
```suggestion
			"call dolt_commit('-am', 'inserting 2 rows at HEAD');",
```

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	},
+	{
+		Name: "dolt_workspace_* deleted row",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"call dolt_commit('-Am', 'creating table t');",
+
+			"insert into tbl values (42,42);",
+			"insert into tbl values (43,43);",
+			"call dolt_commit('-am', 'inserting rows 3 rows at HEAD');",
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go:133]:
```suggestion
			"call dolt_commit('-am', 'inserting 2 rows at HEAD');",
```

File: go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+	},
+	{
+		Name: "dolt_workspace_* clean workspace",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"call dolt_commit('-Am', 'creating table t');",
+
+			"insert into tbl values (42,42);",
+			"insert into tbl values (43,43);",
+			"call dolt_commit('-am', 'inserting rows 3 rows at HEAD');",
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go:162]:
```suggestion
			"call dolt_commit('-am', 'inserting 2 rows at HEAD');",
```

**Review by fulghum — APPROVED** [review]
Looks great! All I noticed was a typo in test commit messages. 😎

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ea1c0db</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ea1c0db</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>48f6bc2</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>48f6bc2</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**